File: 405. Convert a Number to Hexadecimal/convertanumbertohexadecimal.java
class Solution {
    public String toHex(int num) {
        String digits = "0123456789abcdef";
        char[] result = new char[8];

        for (int i = 7; i >= 0; i--) {
            result[i] = digits.charAt(num & 15);
            num >>= 4;
        }

        int start = 0;

        while (result[start] == '0' && start < 7) {
            start++;
        }

        return new String(result, start, 8 - start);
    }
}

File: 405. Convert a Number to Hexadecimal/convertanumbertohexadecimal.py
class Solution:
    def toHex(self, num: int) -> str:
        digits = "0123456789abcdef"
        result = ""

        for i in range(8):
            result = digits[num & 15] + result
            num >>= 4

        start = 0

        while result[start] == '0' and start < 7:
            start += 1

        return result[start:]
